portfolio/src/main/client/src/components/sections/comments.ts

```diff
@@ -0,0 +1,87 @@
+import {htmlElement} from '@src/util/html';
+import '@res/style/sections/comments.scss';
+
+export interface Comment {
+  id: number;
+  name: string;
+  user: string;
+  content: string;
+  upvotes: number;
+  downvotes: number;
+}
+
+/**
+ * Loads comments from server and adds them to the component.
+ * @param el The element to load the comments into.
+ */
+async function load(el: HTMLElement) {
+  try {
+    const response = await fetch('/api/comments');
+    const data = await response.json() as Comment[];
+    const comments = data.map(Comment);
+
+    const listEl = el.getElementsByTagName('ul')[0];
+    while (listEl.firstChild) listEl.firstChild.remove();
+    listEl.append(...comments);
+  } catch (e) {
+    el.append(htmlElement`<li class="error">Failed to load comments.</li>`);
+  }
+}
+
+const Comment = (comment: Comment): HTMLElement => {
+  // user supplied strings cannot be interpolated directly to avoid XSS
+  const content: HTMLElement = htmlElement`<div class="content"></div>`;
+  content.innerText = comment.content;
+
+  const name: HTMLElement = htmlElement`<span class="name"></span>`;
+  name.innerText = comment.name;
+
+  return htmlElement`
+  <li class="comment" data-id="${comment.id.toString()}">
+    <span class="info">
+      ${name}
+      <span class="upvotes">+${comment.upvotes.toString()}</span>
+      <span class="downvotes">-${comment.downvotes.toString()}</span>
+    </span>
+    ${content}
+  </li>`;
+};
+
+export const CommentSection = (): HTMLElement => {
+  const form: HTMLFormElement = htmlElement`
+    <form>
+      <input id="username" name="username" type="text" />
+      <textarea name="content">
+      </textarea>
+      <button type="submit">
+        Comment
+      </button>
+    </form>`;
+
+  // eslint-disable-next-line @typescript-eslint/no-misused-promises
+  form.addEventListener('submit', async (ev) => {
+    ev.preventDefault();
+
+    try {
+      const response = await fetch(
```

Comment:
Instead of using an `await` with a try/catch, you can do:
```
fetch(...).then(...).catch(...);
```

Reply:
I like using try/catch more, though.

Reply:
Using an IIFE when it is not necessary seems like an equally bad, if not worse, solution than adding ESLint disable comments to me. It is more verbose and will puzzle someone reading this code in the future. I could put a comment next to it that says "this is to get rid of the ESLint error", but then why can't I just use a comment to get rid of the ESLint error?

Finally, using callbacks to handle the Promise will lead to control flow becoming more complicated when I add more stuff to this event handler (for example, showing a toast notification, which I also plan to implement using Promises).

What do you think?

Reply:
These are fair points. As a general rule, you shouldn't ever need to disable a linter error when you own the code end to end. In our codebase, the vast majority of disabled lint errors exist to access private or undocumented members of a class whose code we can't refactor (normally external dependencies).

Disabling linter errors tends to be symptomatic of code that could be formatted better. As you mentioned, using an IIFE would reduce readability. I agree. The proper way to handle this, and to handle large callback blocks in general, is to make them their own functions instead of using large anonymous functions:
```
async function fetchComments(form) {
  ...
}

...

form.addEventListener('submit', () => {
  void fetchComments(form);
});
```

Reply:
Makes sense. Refactored it.